test/test_indexmaps.jl

```diff
@@ -69,4 +69,67 @@ Random.seed!(1234)
     xyzvals_approx = calculate_p(s, ind_to_ind_value_map, [i for i in 1:dimension(s)])
     xyzvals ≈ xyzvals_approx
   end
+
+  @testset "test grid_points" begin
+    L = 16
+    base = 2
+    g = named_comb_tree((2, L ÷ 2))
+    s = continuous_siteinds(g; map_dimension=2)
+
+    #first set -- irregular span
+    N = 125
+    a = 0.12
+    b = 0.95
+
+    test_gridpoints = grid_points(s, N, 1; span=[a, b])
+
+    points_in_span = floor(b * base^(L / 2)) - ceil(a * base^(L / 2)) + 1
+    N_gridpoints = Int(floor(points_in_span / ceil(points_in_span / N))) + 1
+
+    @test length(test_gridpoints) == N_gridpoints
+    @test test_gridpoints[1] >= a
+    @test test_gridpoints[end] < b
+    internal = (test_gridpoints[2] - test_gridpoints[1])
+    left = (test_gridpoints[1] - a)
+    right = (b - test_gridpoints[end])
+    @test internal >= left && internal >= right
+
+    #second set
+    N = 32
+    a = 0.25
+    b = 0.5
+    test2 = grid_points(s, N, 1; span=[a, b])
+    @test length(test2) == N
+    @test test2[1] >= a
+    @test test2[end] < b
+    internal = (test2[2] - test2[1])
+    left = (test2[1] - a)
+    right = (b - test2[end])
+    @test internal >= left && internal >= right
```

Comment:
Can each of these commented sets be their own testset?

Reply:
yes